src/config.rs

```diff
@@ -5,13 +9,65 @@ pub trait Config {
 
     fn current_shell_name(&self) -> String;
 
-    fn set_current_shell_name(&mut self, name: &str);
+    fn set_current_shell_name(&mut self, name: &str) -> io::Result<()>;
 
-    fn does_shell_exist(&self, name: &str) -> bool;
+    fn does_shell_exist(&self, name: &str) -> io::Result<bool>;
+}
+
+#[derive(Clone)]
+pub struct FsConfig {
+    pub root_path: PathBuf,
+    pub current_shell: String,
+}
+
+impl FsConfig {
+    fn new(root_path: PathBuf) -> io::Result<Self> {
+        let config_path = root_path.join("current_shell");
+        let config_display = config_path.display();
+
+        let mut file = try!(File::open(&config_path));
+        let mut current_shell = String::new();
+
+        try!(file.read_to_string(&mut current_shell));
+
+        Ok(FsConfig {
+            root_path: root_path,
+            current_shell: current_shell,
+        })
+    }
+}
+
+impl Config for FsConfig {
+    fn root_path(&self) -> &PathBuf {
+        &self.root_path
+    }
+
+    fn current_shell_name(&self) -> String {
+        self.current_shell.clone()
```

Comment:
I think that's something to think about. We should keep our eye on it as we're using it later and decide if that's actually what we want. For now I'd prefer to leave this untouched.